File: scenarios/identity-python-credential-chain/golden/environment_detector.py
from __future__ import annotations

import os


def detect_environment() -> str:
    if any(
        os.getenv(name)
        for name in (
            "CI",
            "TF_BUILD",
            "BUILD_SOURCESDIRECTORY",
            "AZURE_PIPELINE_WORKSPACE",
        )
    ):
        return "ci"
    if os.getenv("IDENTITY_ENDPOINT") or os.getenv("MSI_ENDPOINT"):
        return "production"
    return "dev"

File: scenarios/identity-java-credential-chain/golden/src/main/java/com/example/EnvironmentDetector.java
package com.example;

public final class EnvironmentDetector {
    private EnvironmentDetector() {
    }

    public static DeploymentEnvironment detectEnvironment() {
        if (present("CI")
                || present("TF_BUILD")
                || present("BUILD_SOURCESDIRECTORY")
                || present("AZURE_PIPELINE_WORKSPACE")) {
            return DeploymentEnvironment.CI;
        }
        if (present("IDENTITY_ENDPOINT") || present("MSI_ENDPOINT")) {
            return DeploymentEnvironment.PRODUCTION;
        }
        return DeploymentEnvironment.DEV;
    }

    private static boolean present(String name) {
        String value = System.getenv(name);
        return value != null && !value.isBlank();
    }
}
